src/DataProtection/DataProtection/src/KeyManagement/KeyRingBasedDataProtector.cs

```diff
@@ -381,6 +353,118 @@ public byte[] GetAadForKey(Guid keyId, bool isProtecting)
             }
         }
 
+#if NET10_0_OR_GREATER
```

Comment:
What part is preventing from running on lower TFMs?

Reply:
Agreed, this should be fine to be consistent on all TFMs

Reply:
`System.Numerics.BitOperations.LeadingZeroCount()` and `Encoding.GetBytes(ReadOnlySpan<char>, Span<byte>)` are not available under netstandard2.0 / netFx. 

I have changed the code not to use the previous implementation in whole, but only those APIs.

Reply:
I can show you how to shim the encoding.  For lzcnt; maybe `#if` to the `while` loop approach, i.e. the write me inus the actual writes. The alternative is some math: https://github.com/dotnet/runtime/blob/1d1bf92fcf43aa6981804dc53c5174445069c9e4/src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs#L207

Reply:
I think we can use the existing API on Encoding.GetBytes() which allows to write directly to the byte array (since we have one, it is pretty convenient):

`index += EncodingUtil.SecureUtf8Encoding.GetBytes(purpose, charIndex: 0, charCount: purpose.Length, bytes: targetArr, byteIndex: index);`

let me know if you still want to change it to some other implementation like the one you suggested:
```
public static unsafe int GetBytes(this UTF8Encoding encoding, ReadOnlySpan<char> source, Span<byte> target)
    {
        if (source.IsEmpty) return 0;
        fixed (char* cPtr = source)
        {
            fixed (byte* bPtr = target)
            {
                return encoding.GetBytes(cPtr, source.Length, bPtr, target.Length);
            }
        }
    }
```